### ISSUE TITLE
Pane sync not working

### ISSUE THREAD

**Issue by internationils** [retext-project/retext#513]
Running 7.1.0 with Qt 5.12.8 under latest Ubuntu Focal... the preview pane and source pane are not synchronizing, even though the option is checked. I have tried restarting, but no luck.

**Comment by mitya57** [comment]
What exactly happens — is the preview panel blank, or it displays a rendered content of some old version of the text? Does it happen with any document or only a specific one? Does closing and opening preview help? And what about closing and opening the file?

Also can you please run `retext` from terminal and see if there is any output pasted?

**Comment by internationils** [comment]
No, the preview panel is correct and up to date, but using the scrollbars in either the text or preview does not scroll the other one. If I scroll the preview to the top and go to the bottom of the document, then I can type but the new text does not cause a scroll to the bottom in the preview pane.

When both are at the bottom and I type, then it scolls along with the new text just fine. Maybe add an option to "center on new text" that moves the preview pane to the point where you just changed text?

Also, minor nit: if I have the current line highlight option on, then with text wrapped only the one line not the several lines (i.e. line beginning to EOL) are highlighted in yellow. Not sure if that is intended or not.

**Comment by mitya57** [comment]
The synchronized scroll requires using the WebKit or WebEngine renderer, which you can enable from the Edit menu. Can you check if it is enabled?

Also, you are using Markdown, not reStructuredText, right?

> with text wrapped only the one line not the several lines (i.e. line beginning to EOL) are highlighted in yellow

One of the use cases of this feature is helping to locate the cursor. Highlighting less is better from this point of view. But if you disagree please open a separate bug.

**Comment by internationils** [comment]
Ah, when checking "use WebKit" the scrolling syncs. Might be worth mentioning this next to the preferences option? Feel free to close, thanks!
I've opened https://github.com/retext-project/retext/issues/516 for the highlighting topic.

**Comment by mitya57** [comment]
It is mentioned in the [FAQ](https://github.com/retext-project/retext/wiki/FAQ#can-i-use-synchronized-scrolling), but mentioning it in the Preferences is also a good idea. I will look at it.

**Comment by toolonely** [comment]
Since synced scrolled of panes is a common sense expected feature, perhaps the switch to WebKit should happen automatically when the iuser enables live preview in the menu? I'm on ReText 7.0.4 and live preview was disabled by default. I enabled it but the gotcha about pane sync is not obvious at all, I had to come here and search through the issues.

**Comment by ReddSpark1** [comment]
Same here - glad I found the solution!
